CMakeLists.txt

```diff
@@ -610,6 +640,10 @@ if(CUBEB_FOUND)
 else()
   message(STATUS "Using static cubeb from Externals")
   add_subdirectory(Externals/cubeb EXCLUDE_FROM_ALL)
+  set_target_properties(cubeb PROPERTIES FOLDER Externals)
+  set_target_properties(speex PROPERTIES FOLDER Externals)
+  disable_all_warnings(cubeb)
```

Comment:
Just a note that it's possible to avoid messing with the warnings of externals in most cases, by moving the addition of compilation flags into `Source/CMakeLists.txt`. `add_compile_options` operates on a by-directory basis, and since the compilation flags were moved into the top level CMake file, this means they'll propagate to every target in the source tree, including the externals.

Moving them into the CMakeLists.txt in the Sources directory will apply our desired compilation flags to our sources, and leave the externals' build compilation options alone.

Reply:
The problem is that it will still produce some warnings we don't care about AFAICT.
It is definitely better to force all warnings off for Externals IMO.

Reply:
+1 for moving the options into Source/CMakeLists.txt. All the additional macros/include directories and such aren't needed for compiling the externals, and unnecessarily force them to recompile when said options are changed.

Reply:
Fair enough, good point.